### ISSUE TITLE
Cooldown bar+projectile bugs

### ISSUE THREAD

**Issue by SpinnyBoiii** [apace100/origins-fabric#241]
some bugs in version 0.7.0(tested solo,no idea about the visual effects in multiplayer):

the cooldown bar(at least, for lower cooldown skills(in my case it happened on a 1.5s cooldown which got turned into a 3+ second cooldown)) can sometimes fail in resetting cooldowns, causing it to appear multiple times in a row till it succeeds

most projectiles will visually speed up after a while(and depending on where you're aiming, visually curve)

minecraft:shulker_bullet now immediately despawns on use

exploding projectiles will sometimes explode right before reaching the point they "collided" with, causing no actual destruction

**Comment by SpinnyBoiii** [comment]
something to be added to the projectile pausing: they dont despawn either, checked it with a /kill command after moving like 200+ blocks in the opposite direction of the shot and waiting 30s

**Comment by SpinnyBoiii** [comment]
was doing some more testing and, for the sake of attempting to balance projectiles(despite the visual glitching), i put the projectile speed to 0.1
turns out this will have it be the default speed of that projectile in vanilla minecraft, going any faster will result in the projectile doing a small teleport after the initial startup(which goes at the projectile's basic minecraft speed).
this was tested on V0.7.3 as i'm not currently planning to move my modpack to minecraft 1.17.
so basically: the visual bugging will only exist if the projectile moves faster than it's basic minecraft speed(which seems to always be speed value of 0.1, meaning the current default of 1.5 will always result in it visually glitching)
